### DISCUSSION TITLE
Very large static app

### DISCUSSION THREAD

**Discussion by eecue** [sveltejs/kit#8868] [Q&A] [answered]
I have been building a new version of my photo gallery using SvelteKit and I'm ready to publish it, however I'm running in to some issues with the speed that it's building.  The gallery has close to a million pages total.  I'm using msgpack files (like slim json basically) to store the data, but I don't want to include that data in the generated site.  

If I reference the msgpack files as external URLs to load, the build is quite slow, a few thousand pages per minute. If I reference the files locally in `/static` and build using `adapter-static` it copies all of the msgpack files into the build directory. 

My end goal is to push all the html/js/etc files up to s3 (the images have already been uploaded).  

So I guess I have a few questions:

1) Can I tell SvelteKit to ignore a path in `/static` but still use the data it pulls from there to generate the pages
2) How can I speed up the build process?
3) Ideally I wouldn't build to `./svelte-kit/build` but instead to a path outside of the SvelteKit project folder

**Comment by teemingc** [comment, marked as answer]
1 & 2. You should be able to place these files in `src/lib` and import the filepaths such as
```js
const msgpackFiles = import.meta.glob('$lib/data/*.msp');
```
https://vitejs.dev/guide/features.html#glob-import

3. You should be able to configure the build path (I haven't test this).
https://kit.svelte.dev/docs/adapter-static#options-pages
```js
import adapter from '@sveltejs/adapter-static';
 
export default {
  kit: {
    adapter: adapter({
      pages: '../../somewhere/outside',
    })
  }
};
```

**Reply by eecue** [reply]
Thanks I was able to solve 1 & 2 with your help, however the third one seems to be a [known issue](https://github.com/sveltejs/kit/issues/4052).

**Reply by eecue** [reply]
It's now building about 1k pages per second, thanks!

**Reply by teemingc** [reply]
Otherwise, you can edit the `build` script in `package.json` to look something like `vite build && mv .svelte-kit/build ./../outside`
